components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/common/Util.java

```diff
@@ -56,8 +56,16 @@ private Util() {
      */
     public static StorageSystemManager getStorageSystemManager() {
 
-        return (StorageSystemManager) PrivilegedCarbonContext.getThreadLocalCarbonContext()
-                .getOSGiService(StorageSystemManager.class, null);
+        try {
+            return (StorageSystemManager) PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                    .getOSGiService(StorageSystemManager.class, null);
+        } catch (Throwable e) {
```

Comment:
what is the requirement to catch throwable here?

Reply:
throwable was used here to handle NPE caused in the getOSGiService method. This is cause we are loading the properties file in https://github.com/wso2/identity-media/pull/20/files/db8534c3e3a745f3fd56809c2dc973db897e78be#diff-198e0ade17d75d1c1d73b3b7952e701aR51
If there is an error while reading the file the line: https://github.com/wso2/identity-media/pull/20/files/db8534c3e3a745f3fd56809c2dc973db897e78be#diff-198e0ade17d75d1c1d73b3b7952e701aR59 will not be executed, hence the above NPE which I have mentioned.